R/recipe.R

```diff
@@ -414,20 +424,36 @@ prep.recipe <-
 
     running_info <- x$term_info %>% mutate(number = 0, skip = FALSE)
 
+    needs_tuning <- purrr::map(x$steps, ~ {
```

Comment:
I'm concerned about the performance of this idiom. Could we 1) give the anonymous function a name (`map()` experiences slowdowns with anaonymous functions) and 2) transition `tibble()` to `new_tibble()`?

Reply:
refactored in https://github.com/tidymodels/recipes/pull/1251/commits/c6e946ac0de5410f1491c59c338b78eb46809d44 to use function, and `new_tibble()`